libs/wingsdk/src/ui/file-browser.ts

```diff
@@ -0,0 +1,159 @@
+import { Construct } from "constructs";
+import { VisualComponent } from "./base";
+import { Function } from "../cloud";
+import { fqnForType } from "../constants";
+import { App, UIComponent } from "../core";
+import { IInflight } from "../std";
+
+/**
+ * Global identifier for `FileBrowser`.
+ */
+export const FILE_BROWSER_FQN = fqnForType("ui.FileBrowser");
+
+/**
+ * A file browser can be used to browse files.
+ */
+export class FileBrowser extends VisualComponent {
+  /**
+   * Creates a new ui.FileBrowser instance through the app.
+   * @internal
+   */
+  public static _newFileBrowser(
+    scope: Construct,
+    id: string,
+    label: string,
+    putHandler: IFileBrowserPutHandler,
+    deleteHandler: IFileBrowserDeleteHandler,
+    getHandler: IFileBrowserGetHandler,
+    listHandler: IFileBrowserListHandler
+  ): FileBrowser {
+    return App.of(scope).newAbstract(
+      FILE_BROWSER_FQN,
+      scope,
+      id,
+      label,
+      putHandler,
+      getHandler,
+      deleteHandler,
+      listHandler
+    );
+  }
+
+  private readonly getFn: Function;
+  private readonly putFn: Function;
+  private readonly deleteFn: Function;
+  private readonly listFn: Function;
+  private readonly label: string;
+
+  constructor(
+    scope: Construct,
+    id: string,
+    label: string,
+    putHandler: IFileBrowserPutHandler,
+    deleteHandler: IFileBrowserDeleteHandler,
+    getHandler: IFileBrowserGetHandler,
+    listHandler: IFileBrowserListHandler
```

Comment:
I think these should all be keywords ("props"):

```suggestion
    props: FileBrowserHandlers,
```

And then:

```js
extern interface FileBrowserHandlers {
  putHandler: IFileBrowserPutHandler;
  deleteHandler: IFileBrowserDeleteHandler;
  getHandler: IFileBrowserGetHandler;
  listHandler: IFileBrowserListHandler;
}
```